questions.ts

```diff
@@ -755,35 +755,56 @@ export const askQuestions = async (
   }
 
   if (program.dataSource?.type === "web" && program.framework === "fastapi") {
-    let { baseUrl } = await prompts(
-      {
-        type: "text",
-        name: "baseUrl",
-        message: "Please provide base URL of the website:",
-        initial: "https://www.llamaindex.ai",
-      },
-      handlers,
-    );
-    try {
-      if (!baseUrl.includes("://")) {
-        baseUrl = `https://${baseUrl}`;
-      }
-      const checkUrl = new URL(baseUrl);
-      if (checkUrl.protocol !== "https:" && checkUrl.protocol !== "http:") {
-        throw new Error("Invalid protocol");
+    const validateUrl = (value: string) => {
+      for (let url of value.split(",")) {
+        if (!url.includes("://")) {
+          url = `https://${url}`;
+        }
+        const urlObj = new URL(url);
+        if (urlObj.protocol !== "https:" && urlObj.protocol !== "http:") {
+          return `URL=${url} has invalid protocol, only allow http or https`;
+        }
       }
-    } catch (error) {
-      console.log(
-        red(
-          "Invalid URL provided! Please provide a valid URL (e.g. https://www.llamaindex.ai)",
-        ),
+      return true;
+    };
+
+    program.dataSource.config = [];
+
+    while (true) {
+      const questions: any[] = [
+        {
+          type: "text",
+          name: "baseUrl",
+          message: "Please provide base URL of the website: ",
+          initial: "https://www.llamaindex.ai",
+          validate: validateUrl,
+        },
+        {
+          type: "text",
+          name: "prefix",
+          message:
+            "Please provide prefix for the URLs you would like to crawl: ",
+          initial: "https://www.llamaindex.ai/enterprise",
+          validate: validateUrl,
+        },
+        {
```

Comment:
why do we need two parameters for the URL?

Reply:
It allows users to config more crawler parameters, i ignored the prefix to simplify the cli questions before. If you think it's confusing the user, then we can remove it and consider adding more guidelines once we move to yaml config.

Reply:
yes i think it's better. we already have to many parameters for the initial generation. 
Initial generation should be easy (so user are not leaving it) and then later they can configure and tweak more